components/formats-api/src/loci/formats/in/DynamicMetadataOptions.java

```diff
@@ -476,5 +487,41 @@ public File getFile(String name, File defaultValue) {
     }
     return new File(val);
   }
+  
+  public void loadOptions(File optionsFile, ArrayList<String> availableOptionKeys) throws IOException, FormatException {
+    if (!optionsFile.exists()) {
+      LOGGER.trace("Options file doesn't exist: {}", optionsFile);
+      // TODO: potentially create option
+      return;
+    }
+    if(!optionsFile.canRead()) {
+      LOGGER.trace("Can't read options file: {}", optionsFile);
+      return;
+    }
+
+    IniParser parser = new IniParser();
+    IniList list = parser.parseINI(optionsFile);
+    for (IniTable attrs: list) {
+      for (String key: attrs.keySet()) {
+        if (!availableOptionKeys.contains(key)) {
+          LOGGER.warn("Metadata Option Key is not supported in this reader " + key);
+        }
+        set(key, attrs.get(key));
+      }
+    }
+  }
+
+  public static File getMetadataOptionsFile(String id) {
+    File f = new File(id);
+    if (f != null && f.getParent() != null) {
+      String p = f.getParent();
+      String n = f.getName();
+      if (n.indexOf(".") >= 0) {
+        n = n.substring(0, n.indexOf("."));
```

Comment:
While working on #3617, I realized one downside of this naming pattern is that the option filename is not unique. Thinking of the potential scenarios, I think the two principal ones are:

- filesets where constituting files have the same base name but different extensions e.g. `drosophila.h5, drosophila.xml`
- separate filesets sharing the same basename e.g. `test.tiff, test.png`

I will try and test the first case but we might want to review whether the second case is a blocker. Currently I assume it would mean both filesets would share an option file.